cmd/kured/main.go

```diff
@@ -228,8 +228,8 @@ type RebootBlocker interface {
 // PrometheusBlockingChecker contains info for connecting
 // to prometheus, and can give info about whether a reboot should be blocked
 type PrometheusBlockingChecker struct {
-	// URL to contact prometheus API for checking alerts
-	promURL string
+	// labels to query get  alerts
+	includeLabel map[string]string
```

Comment:
Alternatively, we can change this by splitting this into two blockers, as it's effectively what this is.
It duplicates a bit more code, but the separation of those are very clear, and the logic is easier to follow.

We would have to rename the current type as PrometheusAlertsBlockingChecker, and a new one to PrometheusLabelsBlockingChecker. While it would be easier to show/follow what is blocking, it's also quite a big overhead in terms of code, so I am really on the fence about my own proposition.

I hope I will get a community opinion on this comment ;)

Reply:
I do think both 'blockers' are doing in essence the same thing, that is.. but the implementation is different.. both are getting some input-filter, and returning a blocking-list... 
Yes, it's a bit confusing.. But I was trying to do the most 'pragmatic' thing to do, by opt-ing into existing functionality. 
That is; all alerts are being 'Queried', and then iterated over to get alertNames.. I merely added another 'label' check to each alert.
I think what makes it more readable (as stated into the method itself) would be to create separate functions (filterByRegex and filterByLabels) and let these functions do the specific filtering logic. 

I think adding another blocker, probably will lead into doing another query to the prometheus server...

Reply:
Also, As I was testing/checking/investigating, I stumbled upon the Alert() function of the prometheus client.. I was wondering why are we using the Query() function of the prometheus client if there's also an Alert() function that goes to the /api/v1/alerts and gets all Active Alerts. This might give us an easier way to get alerts, and filter them accordingly.  
what do you guys think @evrardjp @jackfrancis

Reply:
Sounds like an improvement to me!